fw/http_parser.c

```diff
@@ -3260,33 +3277,61 @@ __req_parse_host(TfwHttpReq *req, unsigned char *data, size_t len)
 	}
 
 	__FSM_STATE(Req_I_H_Port) {
-		/* See Req_UriPort processing. */
-		if (unlikely(IS_CRLFWS(c))) {
-			if (!req->host_port)
-				/* Header ended before port was parsed. */
-				return CSTR_NEQ;
-			return __data_off(p);
-		}
 		__fsm_sz = __data_remain(p);
 		__fsm_n = __parse_ulong_ws(p, __data_remain(p), &parser->_acc,
 					   USHRT_MAX);
+
 		switch (__fsm_n) {
+		case CSTR_EQ:
```

Comment:
Why we return CSTR_NEQ if CSTR_EQ returned by __parse_ulong_ws?

Reply:
In case of empty port. `__parse_ulong_ws()` returns size of the string.

`#define CSTR_EQ			0`

Reply:
It seems that we should check that port is zero later. It seems that we catch error on TCP segmentation because of it.

Reply:
Yep, you are right, I must check it only at the end of the parsed string.